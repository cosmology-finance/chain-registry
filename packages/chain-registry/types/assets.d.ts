import type { AssetList } from '@chain-registry/types';
declare const assets: AssetList[];
export default assets;
