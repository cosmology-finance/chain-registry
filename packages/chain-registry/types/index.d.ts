export * from './mainnet';
export * from './testnet';
export * from './noncosmos';
import all from './all';
export default all;
declare const assets: import("@chain-registry/types").AssetList[], chains: import("@chain-registry/types").Chain[], ibc: import("@chain-registry/types").IBCInfo[];
export { assets, chains, ibc };
