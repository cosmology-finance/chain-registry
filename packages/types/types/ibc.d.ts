export interface IBCInfo {
    $schema?: string;
    chain_1: {
        chain_name: string;
        client_id: string;
        connection_id: string;
    };
    chain_2: {
        chain_name: string;
        client_id: string;
        connection_id: string;
    };
    channels: {
        chain_1: {
            channel_id: string;
            client_id?: string;
            connection_id?: string;
            port_id: string;
        };
        chain_2: {
            channel_id: string;
            client_id?: string;
            connection_id?: string;
            port_id: string;
        };
        ordering: string;
        version: string;
        fee_version?: string;
        tags?: object;
    }[];
    /**
     * ibc connection operator information.
     */
    operators?: {
        chain_1: ChainOperatorInfo;
        chain_2: ChainOperatorInfo;
        memo: string;
        /**
         * Operator display name
         */
        name: string;
        discord_handle?: string;
        [k: string]: unknown;
    }[];
}

/**
 * Operator information on a specific chain.
 */
export interface ChainOperatorInfo {
    address?: string;
    [k: string]: unknown;
  }