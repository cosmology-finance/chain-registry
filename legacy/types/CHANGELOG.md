# Change Log

All notable changes to this project will be documented in this file.
See [Conventional Commits](https://conventionalcommits.org) for commit guidelines.

## 0.43.6 (2024-05-29)

**Note:** Version bump only for package @chain-registry/types





## 0.43.5 (2024-05-28)

**Note:** Version bump only for package @chain-registry/types





## 0.43.4 (2024-05-27)

**Note:** Version bump only for package @chain-registry/types





## 0.43.3 (2024-05-26)

**Note:** Version bump only for package @chain-registry/types





## 0.43.2 (2024-05-25)

**Note:** Version bump only for package @chain-registry/types





## 0.43.1 (2024-05-24)

**Note:** Version bump only for package @chain-registry/types





# 0.43.0 (2024-05-24)

**Note:** Version bump only for package @chain-registry/types





# 0.42.0 (2024-05-24)

**Note:** Version bump only for package @chain-registry/types





## 0.41.4 (2024-05-23)

**Note:** Version bump only for package @chain-registry/types





## 0.41.3 (2024-05-22)

**Note:** Version bump only for package @chain-registry/types





## 0.41.2 (2024-05-21)

**Note:** Version bump only for package @chain-registry/types





## 0.41.1 (2024-05-19)

**Note:** Version bump only for package @chain-registry/types





# 0.41.0 (2024-05-18)

**Note:** Version bump only for package @chain-registry/types





# 0.40.0 (2024-05-18)

**Note:** Version bump only for package @chain-registry/types





# 0.39.0 (2024-05-18)

**Note:** Version bump only for package @chain-registry/types





# 0.38.0 (2024-05-18)

**Note:** Version bump only for package @chain-registry/types





# 0.37.0 (2024-05-18)

**Note:** Version bump only for package @chain-registry/types





# 0.36.0 (2024-05-18)

**Note:** Version bump only for package @chain-registry/types





# 0.35.0 (2024-05-18)

**Note:** Version bump only for package @chain-registry/types





# 0.34.0 (2024-05-18)

**Note:** Version bump only for package @chain-registry/types





## 0.33.1 (2024-05-18)

**Note:** Version bump only for package @chain-registry/types





# 0.33.0 (2024-05-18)

**Note:** Version bump only for package @chain-registry/types





# 0.32.0 (2024-05-17)

**Note:** Version bump only for package @chain-registry/types





# 0.31.0 (2024-05-17)

**Note:** Version bump only for package @chain-registry/types





# 0.30.0 (2024-05-17)

**Note:** Version bump only for package @chain-registry/types





## 0.29.2 (2024-05-15)

**Note:** Version bump only for package @chain-registry/types





## 0.29.1 (2024-05-14)

**Note:** Version bump only for package @chain-registry/types





# 0.29.0 (2024-05-08)

**Note:** Version bump only for package @chain-registry/types





## 0.28.12 (2024-05-07)

**Note:** Version bump only for package @chain-registry/types





## 0.28.11 (2024-05-05)

**Note:** Version bump only for package @chain-registry/types





## 0.28.10 (2024-05-04)

**Note:** Version bump only for package @chain-registry/types





## 0.28.9 (2024-05-03)

**Note:** Version bump only for package @chain-registry/types





## 0.28.8 (2024-05-02)

**Note:** Version bump only for package @chain-registry/types





## 0.28.7 (2024-05-01)

**Note:** Version bump only for package @chain-registry/types





## 0.28.6 (2024-05-01)

**Note:** Version bump only for package @chain-registry/types





## 0.28.5 (2024-04-30)

**Note:** Version bump only for package @chain-registry/types





## 0.28.4 (2024-04-29)

**Note:** Version bump only for package @chain-registry/types





## 0.28.3 (2024-04-27)

**Note:** Version bump only for package @chain-registry/types





## 0.28.2 (2024-04-26)

**Note:** Version bump only for package @chain-registry/types





## 0.28.1 (2024-04-25)

**Note:** Version bump only for package @chain-registry/types





# 0.28.0 (2024-04-24)

**Note:** Version bump only for package @chain-registry/types





# 0.27.0 (2024-04-24)

**Note:** Version bump only for package @chain-registry/types





# 0.26.0 (2024-04-24)

**Note:** Version bump only for package @chain-registry/types





## 0.25.8 (2024-04-24)

**Note:** Version bump only for package @chain-registry/types





## 0.25.7 (2024-04-23)

**Note:** Version bump only for package @chain-registry/types





## 0.25.6 (2024-04-21)

**Note:** Version bump only for package @chain-registry/types





## 0.25.5 (2024-04-20)

**Note:** Version bump only for package @chain-registry/types





## 0.25.4 (2024-04-19)

**Note:** Version bump only for package @chain-registry/types





## 0.25.3 (2024-04-18)

**Note:** Version bump only for package @chain-registry/types





## 0.25.2 (2024-04-17)

**Note:** Version bump only for package @chain-registry/types





## 0.25.1 (2024-04-16)

**Note:** Version bump only for package @chain-registry/types





# 0.25.0 (2024-04-14)

**Note:** Version bump only for package @chain-registry/types





## 0.24.1 (2024-04-14)

**Note:** Version bump only for package @chain-registry/types





# 0.24.0 (2024-04-13)

**Note:** Version bump only for package @chain-registry/types





# 0.23.0 (2024-04-13)

**Note:** Version bump only for package @chain-registry/types





# 0.22.0 (2024-04-13)

**Note:** Version bump only for package @chain-registry/types





## 0.21.1 (2024-04-12)

**Note:** Version bump only for package @chain-registry/types





# 0.21.0 (2024-04-11)

**Note:** Version bump only for package @chain-registry/types





# 0.20.0 (2024-04-11)

**Note:** Version bump only for package @chain-registry/types





# 0.19.0 (2024-04-11)

**Note:** Version bump only for package @chain-registry/types





## 0.18.19 (2024-04-11)

**Note:** Version bump only for package @chain-registry/types





## 0.18.18 (2024-04-10)

**Note:** Version bump only for package @chain-registry/types





## 0.18.17 (2024-04-09)

**Note:** Version bump only for package @chain-registry/types





## 0.18.16 (2024-04-06)

**Note:** Version bump only for package @chain-registry/types





## 0.18.15 (2024-04-05)

**Note:** Version bump only for package @chain-registry/types





## 0.18.14 (2024-04-04)

**Note:** Version bump only for package @chain-registry/types





## 0.18.13 (2024-04-03)

**Note:** Version bump only for package @chain-registry/types





## 0.18.12 (2024-04-02)

**Note:** Version bump only for package @chain-registry/types





## 0.18.11 (2024-03-31)

**Note:** Version bump only for package @chain-registry/types





## 0.18.10 (2024-03-30)

**Note:** Version bump only for package @chain-registry/types





## 0.18.9 (2024-03-29)

**Note:** Version bump only for package @chain-registry/types





## 0.18.8 (2024-03-28)

**Note:** Version bump only for package @chain-registry/types





## [0.18.7](https://github.com/cosmology-tech/chain-registry/compare/@chain-registry/types@0.18.6...@chain-registry/types@0.18.7) (2024-03-27)

**Note:** Version bump only for package @chain-registry/types





## [0.18.6](https://github.com/cosmology-tech/chain-registry/compare/@chain-registry/types@0.18.5...@chain-registry/types@0.18.6) (2024-03-27)

**Note:** Version bump only for package @chain-registry/types





## [0.18.5](https://github.com/cosmology-tech/chain-registry/compare/@chain-registry/types@0.18.4...@chain-registry/types@0.18.5) (2024-03-21)

**Note:** Version bump only for package @chain-registry/types





## 0.18.4 (2024-03-21)

**Note:** Version bump only for package @chain-registry/types





## [0.18.3](https://github.com/cosmology-tech/chain-registry/compare/@chain-registry/types@0.18.2...@chain-registry/types@0.18.3) (2024-03-20)

**Note:** Version bump only for package @chain-registry/types





## 0.18.2 (2024-03-20)

**Note:** Version bump only for package @chain-registry/types





## [0.18.1](https://github.com/cosmology-tech/chain-registry/compare/@chain-registry/types@0.18.0...@chain-registry/types@0.18.1) (2024-02-27)

**Note:** Version bump only for package @chain-registry/types





# [0.18.0](https://github.com/cosmology-tech/chain-registry/compare/@chain-registry/types@0.17.1...@chain-registry/types@0.18.0) (2024-01-25)

**Note:** Version bump only for package @chain-registry/types





## [0.17.1](https://github.com/cosmology-tech/chain-registry/compare/@chain-registry/types@0.17.0...@chain-registry/types@0.17.1) (2023-12-15)

**Note:** Version bump only for package @chain-registry/types





# [0.17.0](https://github.com/cosmology-tech/chain-registry/compare/@chain-registry/types@0.16.0...@chain-registry/types@0.17.0) (2023-09-15)

**Note:** Version bump only for package @chain-registry/types





# [0.16.0](https://github.com/cosmology-tech/chain-registry/compare/@chain-registry/types@0.15.0...@chain-registry/types@0.16.0) (2023-04-20)

**Note:** Version bump only for package @chain-registry/types





# [0.15.0](https://github.com/cosmology-tech/chain-registry/compare/@chain-registry/types@0.14.0...@chain-registry/types@0.15.0) (2023-04-18)

**Note:** Version bump only for package @chain-registry/types





# [0.14.0](https://github.com/cosmology-tech/chain-registry/compare/@chain-registry/types@0.13.1...@chain-registry/types@0.14.0) (2022-12-08)

**Note:** Version bump only for package @chain-registry/types





## [0.13.1](https://github.com/cosmology-tech/chain-registry/compare/@chain-registry/types@0.13.0...@chain-registry/types@0.13.1) (2022-10-27)

**Note:** Version bump only for package @chain-registry/types





# [0.13.0](https://github.com/cosmology-tech/chain-registry/compare/@chain-registry/types@0.12.0...@chain-registry/types@0.13.0) (2022-10-20)

**Note:** Version bump only for package @chain-registry/types





# [0.12.0](https://github.com/cosmology-tech/chain-registry/compare/@chain-registry/types@0.11.1...@chain-registry/types@0.12.0) (2022-10-20)

**Note:** Version bump only for package @chain-registry/types





## [0.11.1](https://github.com/cosmology-tech/chain-registry/compare/@chain-registry/types@0.11.0...@chain-registry/types@0.11.1) (2022-10-20)

**Note:** Version bump only for package @chain-registry/types





# [0.11.0](https://github.com/cosmology-tech/chain-registry/compare/@chain-registry/types@0.10.2...@chain-registry/types@0.11.0) (2022-09-20)

**Note:** Version bump only for package @chain-registry/types





## [0.10.2](https://github.com/cosmology-tech/chain-registry/compare/@chain-registry/types@0.10.1...@chain-registry/types@0.10.2) (2022-09-19)

**Note:** Version bump only for package @chain-registry/types





## [0.10.1](https://github.com/cosmology-tech/chain-registry/compare/@chain-registry/types@0.10.0...@chain-registry/types@0.10.1) (2022-09-19)

**Note:** Version bump only for package @chain-registry/types





# [0.10.0](https://github.com/cosmology-tech/chain-registry/compare/@chain-registry/types@0.9.0...@chain-registry/types@0.10.0) (2022-09-19)

**Note:** Version bump only for package @chain-registry/types





# [0.9.0](https://github.com/cosmology-tech/chain-registry/compare/@chain-registry/types@0.8.0...@chain-registry/types@0.9.0) (2022-09-16)

**Note:** Version bump only for package @chain-registry/types





# [0.8.0](https://github.com/cosmology-tech/chain-registry/compare/@chain-registry/types@0.7.1...@chain-registry/types@0.8.0) (2022-09-14)

**Note:** Version bump only for package @chain-registry/types





## [0.7.1](https://github.com/cosmology-tech/chain-registry/compare/@chain-registry/types@0.7.0...@chain-registry/types@0.7.1) (2022-09-08)

**Note:** Version bump only for package @chain-registry/types





# [0.7.0](https://github.com/cosmology-tech/chain-registry/compare/@chain-registry/types@0.6.0...@chain-registry/types@0.7.0) (2022-09-06)

**Note:** Version bump only for package @chain-registry/types





# [0.6.0](https://github.com/cosmology-tech/chain-registry/compare/@chain-registry/types@0.5.0...@chain-registry/types@0.6.0) (2022-08-23)

**Note:** Version bump only for package @chain-registry/types





# [0.5.0](https://github.com/cosmology-tech/chain-registry/compare/@chain-registry/types@0.4.0...@chain-registry/types@0.5.0) (2022-08-20)

**Note:** Version bump only for package @chain-registry/types





# [0.4.0](https://github.com/cosmology-tech/chain-registry/compare/@chain-registry/types@0.3.1...@chain-registry/types@0.4.0) (2022-08-14)

**Note:** Version bump only for package @chain-registry/types





## [0.3.1](https://github.com/cosmology-tech/chain-registry/compare/@chain-registry/types@0.3.0...@chain-registry/types@0.3.1) (2022-08-13)

**Note:** Version bump only for package @chain-registry/types





# [0.3.0](https://github.com/cosmology-tech/chain-registry/compare/@chain-registry/types@0.2.1...@chain-registry/types@0.3.0) (2022-08-13)

**Note:** Version bump only for package @chain-registry/types





## 0.2.1 (2022-08-13)

**Note:** Version bump only for package @chain-registry/types
