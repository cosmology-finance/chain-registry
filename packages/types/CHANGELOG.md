# Change Log

All notable changes to this project will be documented in this file.
See [Conventional Commits](https://conventionalcommits.org) for commit guidelines.

# 0.22.0 (2024-04-12)

**Note:** Version bump only for package @chain-registry/types





# 0.21.0 (2024-04-11)

**Note:** Version bump only for package @chain-registry/types





# 0.20.0 (2024-04-11)

**Note:** Version bump only for package @chain-registry/types





# 0.19.0 (2024-04-11)

**Note:** Version bump only for package @chain-registry/types





## 0.18.19 (2024-04-11)

**Note:** Version bump only for package @chain-registry/types





## 0.18.18 (2024-04-10)

**Note:** Version bump only for package @chain-registry/types





## 0.18.17 (2024-04-09)

**Note:** Version bump only for package @chain-registry/types





## 0.18.16 (2024-04-06)

**Note:** Version bump only for package @chain-registry/types





## 0.18.15 (2024-04-05)

**Note:** Version bump only for package @chain-registry/types





## 0.18.14 (2024-04-04)

**Note:** Version bump only for package @chain-registry/types





## 0.18.13 (2024-04-03)

**Note:** Version bump only for package @chain-registry/types





## 0.18.12 (2024-04-02)

**Note:** Version bump only for package @chain-registry/types





## 0.18.11 (2024-03-31)

**Note:** Version bump only for package @chain-registry/types





## 0.18.10 (2024-03-30)

**Note:** Version bump only for package @chain-registry/types





## 0.18.9 (2024-03-29)

**Note:** Version bump only for package @chain-registry/types





## 0.18.8 (2024-03-28)

**Note:** Version bump only for package @chain-registry/types





## [0.18.7](https://github.com/cosmology-tech/chain-registry/compare/@chain-registry/types@0.18.6...@chain-registry/types@0.18.7) (2024-03-27)

**Note:** Version bump only for package @chain-registry/types





## [0.18.6](https://github.com/cosmology-tech/chain-registry/compare/@chain-registry/types@0.18.5...@chain-registry/types@0.18.6) (2024-03-27)

**Note:** Version bump only for package @chain-registry/types





## [0.18.5](https://github.com/cosmology-tech/chain-registry/compare/@chain-registry/types@0.18.4...@chain-registry/types@0.18.5) (2024-03-21)

**Note:** Version bump only for package @chain-registry/types





## 0.18.4 (2024-03-21)

**Note:** Version bump only for package @chain-registry/types





## [0.18.3](https://github.com/cosmology-tech/chain-registry/compare/@chain-registry/types@0.18.2...@chain-registry/types@0.18.3) (2024-03-20)

**Note:** Version bump only for package @chain-registry/types





## 0.18.2 (2024-03-20)

**Note:** Version bump only for package @chain-registry/types





## [0.18.1](https://github.com/cosmology-tech/chain-registry/compare/@chain-registry/types@0.18.0...@chain-registry/types@0.18.1) (2024-02-27)

**Note:** Version bump only for package @chain-registry/types





# [0.18.0](https://github.com/cosmology-tech/chain-registry/compare/@chain-registry/types@0.17.1...@chain-registry/types@0.18.0) (2024-01-25)

**Note:** Version bump only for package @chain-registry/types





## [0.17.1](https://github.com/cosmology-tech/chain-registry/compare/@chain-registry/types@0.17.0...@chain-registry/types@0.17.1) (2023-12-15)

**Note:** Version bump only for package @chain-registry/types





# [0.17.0](https://github.com/cosmology-tech/chain-registry/compare/@chain-registry/types@0.16.0...@chain-registry/types@0.17.0) (2023-09-15)

**Note:** Version bump only for package @chain-registry/types





# [0.16.0](https://github.com/cosmology-tech/chain-registry/compare/@chain-registry/types@0.15.0...@chain-registry/types@0.16.0) (2023-04-20)

**Note:** Version bump only for package @chain-registry/types





# [0.15.0](https://github.com/cosmology-tech/chain-registry/compare/@chain-registry/types@0.14.0...@chain-registry/types@0.15.0) (2023-04-18)

**Note:** Version bump only for package @chain-registry/types





# [0.14.0](https://github.com/cosmology-tech/chain-registry/compare/@chain-registry/types@0.13.1...@chain-registry/types@0.14.0) (2022-12-08)

**Note:** Version bump only for package @chain-registry/types





## [0.13.1](https://github.com/cosmology-tech/chain-registry/compare/@chain-registry/types@0.13.0...@chain-registry/types@0.13.1) (2022-10-27)

**Note:** Version bump only for package @chain-registry/types





# [0.13.0](https://github.com/cosmology-tech/chain-registry/compare/@chain-registry/types@0.12.0...@chain-registry/types@0.13.0) (2022-10-20)

**Note:** Version bump only for package @chain-registry/types





# [0.12.0](https://github.com/cosmology-tech/chain-registry/compare/@chain-registry/types@0.11.1...@chain-registry/types@0.12.0) (2022-10-20)

**Note:** Version bump only for package @chain-registry/types





## [0.11.1](https://github.com/cosmology-tech/chain-registry/compare/@chain-registry/types@0.11.0...@chain-registry/types@0.11.1) (2022-10-20)

**Note:** Version bump only for package @chain-registry/types





# [0.11.0](https://github.com/cosmology-tech/chain-registry/compare/@chain-registry/types@0.10.2...@chain-registry/types@0.11.0) (2022-09-20)

**Note:** Version bump only for package @chain-registry/types





## [0.10.2](https://github.com/cosmology-tech/chain-registry/compare/@chain-registry/types@0.10.1...@chain-registry/types@0.10.2) (2022-09-19)

**Note:** Version bump only for package @chain-registry/types





## [0.10.1](https://github.com/cosmology-tech/chain-registry/compare/@chain-registry/types@0.10.0...@chain-registry/types@0.10.1) (2022-09-19)

**Note:** Version bump only for package @chain-registry/types





# [0.10.0](https://github.com/cosmology-tech/chain-registry/compare/@chain-registry/types@0.9.0...@chain-registry/types@0.10.0) (2022-09-19)

**Note:** Version bump only for package @chain-registry/types





# [0.9.0](https://github.com/cosmology-tech/chain-registry/compare/@chain-registry/types@0.8.0...@chain-registry/types@0.9.0) (2022-09-16)

**Note:** Version bump only for package @chain-registry/types





# [0.8.0](https://github.com/cosmology-tech/chain-registry/compare/@chain-registry/types@0.7.1...@chain-registry/types@0.8.0) (2022-09-14)

**Note:** Version bump only for package @chain-registry/types





## [0.7.1](https://github.com/cosmology-tech/chain-registry/compare/@chain-registry/types@0.7.0...@chain-registry/types@0.7.1) (2022-09-08)

**Note:** Version bump only for package @chain-registry/types





# [0.7.0](https://github.com/cosmology-tech/chain-registry/compare/@chain-registry/types@0.6.0...@chain-registry/types@0.7.0) (2022-09-06)

**Note:** Version bump only for package @chain-registry/types





# [0.6.0](https://github.com/cosmology-tech/chain-registry/compare/@chain-registry/types@0.5.0...@chain-registry/types@0.6.0) (2022-08-23)

**Note:** Version bump only for package @chain-registry/types





# [0.5.0](https://github.com/cosmology-tech/chain-registry/compare/@chain-registry/types@0.4.0...@chain-registry/types@0.5.0) (2022-08-20)

**Note:** Version bump only for package @chain-registry/types





# [0.4.0](https://github.com/cosmology-tech/chain-registry/compare/@chain-registry/types@0.3.1...@chain-registry/types@0.4.0) (2022-08-14)

**Note:** Version bump only for package @chain-registry/types





## [0.3.1](https://github.com/cosmology-tech/chain-registry/compare/@chain-registry/types@0.3.0...@chain-registry/types@0.3.1) (2022-08-13)

**Note:** Version bump only for package @chain-registry/types





# [0.3.0](https://github.com/cosmology-tech/chain-registry/compare/@chain-registry/types@0.2.1...@chain-registry/types@0.3.0) (2022-08-13)

**Note:** Version bump only for package @chain-registry/types





## 0.2.1 (2022-08-13)

**Note:** Version bump only for package @chain-registry/types
