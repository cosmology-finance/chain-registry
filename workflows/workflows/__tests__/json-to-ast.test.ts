import generate from '@babel/generator';
import * as t from '@babel/types';

import { createAliasedImportDeclaration, createExportAllDeclarations, createExportedArrayDeclaration, createImportDeclaration, createImportsAndExportsAtRootForIndex, createTypedVariableDeclaration, createVariableAssignment, generateImportExportsAtRoot, generateImportsAndExport, JsonObject } from '../src';
const jsonData: JsonObject = {
  $schema: '../assetlist.schema.json',
  chainName: '8ball',
  assets: [
    {
      description: 'The native staking token of 8ball.',
      denomUnits: [
        { denom: 'uebl', exponent: 0 },
        { denom: 'ebl', exponent: 6 }
      ],
      base: 'uebl',
      name: '8ball',
      display: 'ebl',
      'some/property': 'goes-here',
      _prop: 'goes-here',
      symbol: 'EBL',
      logoURIs: {
        png: 'https://raw.githubusercontent.com/cosmos/chain-registry/master/8ball/images/8ball.png',
        svg: 'https://raw.githubusercontent.com/cosmos/chain-registry/master/8ball/images/8ball.svg'
      },
      coingeckoId: '',
      images: [
        {
          png: 'https://raw.githubusercontent.com/cosmos/chain-registry/master/8ball/images/8ball.png',
          svg: 'https://raw.githubusercontent.com/cosmos/chain-registry/master/8ball/images/8ball.svg'
        }
      ]
    }
  ]
};

it('json-to-ast var declaration', () => {
  const ast = createTypedVariableDeclaration('assetList', 'AssetList', jsonData);
  const code = generate(ast).code;
  expect(code).toMatchSnapshot();
});

it('should generate correct ast for JSON and evaluate it', () => {
  const ast = createVariableAssignment('assets', jsonData);
  const code = generate(ast).code;
  let assets;
  eval(code);
  expect(assets).toEqual(jsonData);
});


it('createImportDeclaration', () => {
  const ast = createImportDeclaration(['AssetList', 'Chain'], '@chain-registry/types');
  const code = generate(ast).code;
  expect(code).toMatchSnapshot();
});

it('createAliasedImportDeclaration', () => {
  const ast = createAliasedImportDeclaration('assetList', '_mychain', 'mychain');
  const code = generate(ast).code;
  expect(code).toMatchSnapshot();
});

it('createExportedArrayDeclaration', () => {
  const ast = createExportedArrayDeclaration('assetLists', 'AssetList', ['_bitcannadev1', '_celestiadevnet2', '_interwebdevnet']);
  const code = generate(ast).code;
  expect(code).toMatchSnapshot();
});

it('generateImportsAndExport', () => {
  const [imports, exports] = generateImportsAndExport([
    {
      source: './asset-lists',
      specifier: 'assetLists'
    },
    {
      source: './chains',
      specifier: 'chains'
    }
  ]);
  const code = generate(t.program([
    ...imports,
    exports
  ])).code;
  expect(code).toMatchSnapshot();
});

it('createExportAllDeclarations', () => {
  const imports = createExportAllDeclarations([
    './asset-lists',
    './chains',
    './ibc-data'
  ]);
  const code = generate(t.program([
    ...imports
  ])).code;
  expect(code).toMatchSnapshot();
});

it('generateImportExportsAtRoot', () => {
  const imports = generateImportExportsAtRoot('ibcData', [
    {
      alias: '_mainnet',
      source: './mainnet'
    },
    {
      alias: '_noncosmos',
      source: './noncosmos'
    },
    {
      alias: '_testnets',
      source: './testnets'
    }
  ], 'ibcData', 'IBCData');
  const code = generate(t.program([
    ...imports
  ])).code;
  expect(code).toMatchSnapshot();
});

it('createImportsAndExportsAtRootForIndex', () => {
  const imports = createImportsAndExportsAtRootForIndex([
    { specifier: 'assetLists', source: './asset-lists' },
    { specifier: 'chains', source: './chains' },
    { specifier: 'ibcData', source: './ibc-data' }
  ]);
  const code = generate(t.program([
    ...imports
  ])).code;
  expect(code).toMatchSnapshot();
});