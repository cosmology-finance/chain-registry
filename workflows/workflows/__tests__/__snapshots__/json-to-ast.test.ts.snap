// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`createAliasedImportDeclaration 1`] = `"import { assetList as _mychain } from "mychain";"`;

exports[`createExportAllDeclarations 1`] = `
"export * from "./asset-lists";
export * from "./chains";
export * from "./ibc-data";"
`;

exports[`createExportedArrayDeclaration 1`] = `"export const assetLists: AssetList[] = [_bitcannadev1, _celestiadevnet2, _interwebdevnet];"`;

exports[`createImportDeclaration 1`] = `"import { AssetList, Chain } from "@chain-registry/types";"`;

exports[`createImportsAndExportsAtRootForIndex 1`] = `
"import assetLists from "./asset-lists";
import chains from "./chains";
import ibcData from "./ibc-data";
export default {
  assetLists,
  chains,
  ibcData
};
export { assetLists, chains, ibcData };"
`;

exports[`generateImportExportsAtRoot 1`] = `
"import { ibcData as _mainnet } from "./mainnet";
import { ibcData as _noncosmos } from "./noncosmos";
import { ibcData as _testnets } from "./testnets";
const ibcData: IBCData[] = [..._mainnet, ..._noncosmos, ..._testnets];"
`;

exports[`generateImportsAndExport 1`] = `
"import { assetLists } from "./asset-lists";
import { chains } from "./chains";
export { assetLists, chains };"
`;

exports[`json-to-ast var declaration 1`] = `
"export const assetList: AssetList = {
  $schema: "../assetlist.schema.json",
  chainName: "8ball",
  assets: [{
    description: "The native staking token of 8ball.",
    denomUnits: [{
      denom: "uebl",
      exponent: 0
    }, {
      denom: "ebl",
      exponent: 6
    }],
    base: "uebl",
    name: "8ball",
    display: "ebl",
    "some/property": "goes-here",
    _prop: "goes-here",
    symbol: "EBL",
    logoURIs: {
      png: "https://raw.githubusercontent.com/cosmos/chain-registry/master/8ball/images/8ball.png",
      svg: "https://raw.githubusercontent.com/cosmos/chain-registry/master/8ball/images/8ball.svg"
    },
    coingeckoId: "",
    images: [{
      png: "https://raw.githubusercontent.com/cosmos/chain-registry/master/8ball/images/8ball.png",
      svg: "https://raw.githubusercontent.com/cosmos/chain-registry/master/8ball/images/8ball.svg"
    }]
  }]
};"
`;
